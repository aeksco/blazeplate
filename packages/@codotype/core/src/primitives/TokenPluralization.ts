import { EMPTY_TOKEN_CASING, TokenCasing, TokenPluralization } from "../token";
import { buildTokenPluralization } from "../util";

// // // //

interface TokenPluralizationBuilderParams {
    titleSingular: string;
}

export class TokenPluralizationBuilder implements TokenPluralization {
    plural: TokenCasing = {
        ...EMPTY_TOKEN_CASING,
    };
    singular: TokenCasing = {
        ...EMPTY_TOKEN_CASING,
    };

    constructor(params: TokenPluralizationBuilderParams) {
        const tokenPluralization: TokenPluralization = buildTokenPluralization(
            params.titleSingular,
        );
    }
}
