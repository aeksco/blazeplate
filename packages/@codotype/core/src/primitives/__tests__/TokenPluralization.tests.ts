import { TokenPluralizationBuilder } from "../TokenPluralization";

// // // //

describe("Primitives.TokenPluralization", () => {
    test("works", () => {
        const TokenPluralization = new TokenPluralizationBuilder({
            titleSingular: "User",
        });

        expect(TokenPluralization).toMatchSnapshot();
    });
});
