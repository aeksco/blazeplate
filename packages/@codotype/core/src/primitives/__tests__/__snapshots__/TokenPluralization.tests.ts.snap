// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primatives.TokenPluralization works 1`] = `
TokenPluralizationBuilder {
  "plural": Object {
    "camel": "",
    "kebab": "",
    "pascal": "",
    "snake": "",
    "title": "",
  },
  "singular": Object {
    "camel": "",
    "kebab": "",
    "pascal": "",
    "snake": "",
    "title": "",
  },
}
`;

exports[`Primitives.TokenPluralization works 1`] = `
TokenPluralizationBuilder {
  "plural": Object {
    "camel": "",
    "kebab": "",
    "pascal": "",
    "snake": "",
    "title": "",
  },
  "singular": Object {
    "camel": "",
    "kebab": "",
    "pascal": "",
    "snake": "",
    "title": "",
  },
}
`;
